internal/mode/static/nginx/config/stream_servers_template.go

```diff
@@ -3,13 +3,18 @@ package config
 const streamServersTemplateText = `
 {{- range $s := .Servers }}
 server {
-	{{- if or ($.IPFamily.IPv4) ($s.IsSocket) }}
+	{{- if and ($.IPFamily.IPv4) (not $s.IsSocket) }}
```

Comment:
No, I think it is the opposite. 

We only set `proxy-protocol` if we are listening on socket for stream

```

# is a socket
server {
    listen unix:/var/run/nginx/cafe.example.com-8443.sock proxy_protocol;
	
    # This should be set to whatever the user has configured
	set_real_ip_from 0.0.0.0/0; 
    proxy_pass backend1;
}

# not a socket
server {
    listen 8443;
    listen [::]:8443;
	
    pass $dest8443;
    ssl_preread on;       
}
```

Added this change for cases when it is IPv4 and not socket, we don't use `proxy-protocol` otherwise we do .

Reply:
Apologies, I got it backwards. Regardless, you are already checking for this when you create the RewriteClientIP object on the server. You don't need to check for it here.